### ISSUE TITLE
Create a suite of probability distributions

### ISSUE THREAD

**Issue by buddha314** [buddha314/numsuch#36]
Follow the format of [rpois](https://stat.ethz.ch/R-manual/R-devel/library/stats/html/Poisson.html) in R for multiple, useful distributions.  Please include tests.

**Comment by buddha314** [comment]
Ones I need right off the bat are Poisson, Normal and Beta-Binomial.  Probably Binomial and Multi-Nomial will come quickly.

**Comment by buddha314** [comment]
Looks like the Gamma distribution will need the help of the [tgamma](https://chapel-lang.org/docs/latest/modules/standard/Math.html?highlight=gamma#Math.tgamma) function in Chapel.  And we need that to find the quantiles of the Poisson distribution.  Rabbit hole is getting deep.

**Comment by ben-albrecht** [comment]
Are you opposed to calling out to [GSL](https://www.gnu.org/software/gsl/manual/html_node/Gamma-Functions.html) ?

**Comment by buddha314** [comment]
Nope, but I have no experience doing that, either.  I intend to use the Gamma function to create the [Gamma Distribution](https://en.wikipedia.org/wiki/Gamma_distribution) like in [R](https://stat.ethz.ch/R-manual/R-devel/library/stats/html/GammaDist.html)
